src/TurbulenceClosures/turbulence_closure_implementations/Smagorinskys/smagorinsky.jl

```diff
@@ -0,0 +1,147 @@
+using Oceananigans.AbstractOperations: Average
+using Oceananigans.Fields: FieldBoundaryConditions
+using Oceananigans.Utils: launch!, IterationInterval
+
+using Adapt
+
+using ..TurbulenceClosures:
+    AbstractScalarDiffusivity,
+    ThreeDimensionalFormulation,
+    ExplicitTimeDiscretization,
+    convert_diffusivity
+
+import Oceananigans.Utils: with_tracers
+
+import ..TurbulenceClosures:
+    viscosity,
+    diffusivity,
+    κᶠᶜᶜ,
+    κᶜᶠᶜ,
+    κᶜᶜᶠ,
+    compute_diffusivities!,
+    DiffusivityFields,
+    tracer_diffusivities
+
+#####
+##### The turbulence closure proposed by Smagorinsky and Lilly.
+#####
+
+# struct LagrangianAveragedCoefficient end
+
+struct Smagorinsky{TD, C, P} <: AbstractScalarDiffusivity{TD, ThreeDimensionalFormulation, 2}
```

Comment:
```suggestion
struct Smagorinsky{C, P} <: AbstractScalarDiffusivity{<:ExplicitTimeDisretization, ThreeDimensionalFormulation, 2}
```
I don't think there is an implicit version set up

Reply:
It can use vertically implicit for the diffusion coefficient (like any abstract scalar diffusivity)

Reply:
Ok, I guess we can keep that time-stepping option

Reply:
I think we want to keep support for vertically implicit for wall resolved LES etc